File: C++/118.cpp
class Solution {
public:
    /* Version 1
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res(numRows);
        if(numRows == 0) return res;// Could not return NULL; 
        res[0].push_back(1);
        for(auto i = 1; i < numRows; i++){
            res[i].push_back(1);
            for(auto j = 0; j < res[i-1].size() - 1; j++)
                res[i].push_back(res[i-1][j] + res[i-1][j+1]);
            res[i].push_back(1);
        }
        return res; 
    }
    */
    
    // Version 2    Concise
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res(numRows);
        for(auto i = 0; i < numRows; i++){
            res[i].resize(i+1); // resize() to build each vector
            res[i][0] = 1;
            for(auto j = 1; j < i; j++)
                res[i][j] = res[i-1][j-1] + res[i-1][j];
            res[i][i] = 1;
        }
        return res;
    }
};

File: LC_Python/118.py
class Solution:
    # @param {integer} numRows
    # @return {integer[][]}
    """Version 1
    def generate(self, numRows):
        if not numRows:
            return []
        pt = [[1]]
        for i in range(numRows)[1:]:
            pt.append([1])
            for j in range(i+1)[1:i]:
                pt[i].append(pt[i-1][j-1] + pt[i-1][j])
            pt[i].append(1)
        return pt
    """
    
    def generate(self, numRows):
        pt = []
        for i in range(numRows):
             curr = [1] if i else []
             for j in range(1,i):
                 curr.append(pt[-1][j-1] + pt[-1][j])
             curr.append(1)
             pt.append(curr)
        return pt